### PR TITLE
Automatically build and publish a Docker image

### PR DESCRIPTION
**Pull request by lol768**
ref https://github.com/UniversityofWarwick/tabula/pull/24

### CODE REVIEW

File: .github/workflows/docker-image.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+      env:
+       HUB_LOGIN: "${{ secrets.HUB_LOGIN }}"
+    - run: |
+        IMAGE_NAME="universityofwarwick/sso-stub:latest"
+        ./buildDocker.sh
+        docker image tag sso-stub:latest ${IMAGE_NAME}
+        echo "${HUB_LOGIN}" | docker login -u adamlwilliams --password-stdin
```
Comment (halfninja) [inline .github/workflows/docker-image.yml:19]: Should we be logging in as your account here?

Comment (lol768) [reply .github/workflows/docker-image.yml:19]: Unless you have an alternative, yeah.. (it's not used for anything other than this)

It's publishing to an org but there aren't org-specific tokens.

Comment (lol768) [reply .github/workflows/docker-image.yml:19]: I'd love to use GitHub's but it's got a bunch of its own problems

- They have an old per-repo registry: Can't be used to pull images unauthed
- The new org-based container registries: Can't have images pushed into them using the GitHub Actions access token, personal access tokens require access to all private repositories, no way to scope down to just this

File: .github/workflows/docker-image.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+on:
+  push:
+    branches: [ master ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
```
Comment (halfninja) [inline .github/workflows/docker-image.yml]:
```suggestion
    - name: Checkout, build and push
```

**Review by halfninja — APPROVED** [review]

**Pull request merged**